scripts/core/editor3/components/Editor3.jsx

```diff
@@ -217,10 +234,48 @@ export class Editor3Component extends React.Component {
     handleBeforeInput(chars) {
         const {editorState, onChange, suggestingMode, onCreateAddSuggestion} = this.props;
 
+        if (!this.allowEditSuggestion('insert')) {
+            return 'handled';
+        }
+
         if (suggestingMode) {
             onCreateAddSuggestion(chars);
             return 'handled';
-        } else if (chars !== ' ') {
+        } else if (!this.allowEditSuggestion('backspace')) {
+            // there is a suggestion before the current position -> prevent the copy of
+            // suggestion entity and style
+            const styles = ['BOLD', 'ITALIC', 'UNDERLINE'];
+            const inlineStyle = editorState.getCurrentInlineStyle();
+            const selection = editorState.getSelection();
+            let newSelection = selection.merge({
+                anchorOffset: selection.getStartOffset(),
+                focusOffset: selection.getEndOffset() + chars.length,
+                isBackward: false
+            });
+            let newContentState = editorState.getCurrentContent();
+            let newEditorState;
+
+            newContentState = Modifier.insertText(newContentState, selection, chars);
+            inlineStyle.forEach((style) => {
+                if (styles.indexOf(style) !== -1) {
```

Comment:
You should exclude the styles you don't want instead of including the ones you want. If `strikethrough` or any other style gets added, we'd have a bug here.

Reply:
On user interface we support only the list of styles from 'styles' list.

Reply:
That's fair, but then try to use the same list so it doesn't go out of sync. For example there is a list at `/superdesk-client-core/scripts/core/editor3/components/toolbar/InlineStyleButtons.jsx` which has no connection to your `styles` variable.